addons/event_crm/models/event_lead_rule.py

```diff
@@ -185,6 +185,30 @@ def _run_on_registrations(self, registrations):
 
         return self.env['crm.lead'].create(lead_vals_list)
 
+    def action_execute_rule(self):
+        events = self.event_id or self.env['event.event'].search([('is_finished', '!=', True)])
+        all_registrations = events.mapped('registration_ids')
+        leads_created = self._run_on_registrations(all_registrations)
+        if leads_created:
+            base_message = _("Yee-ha, %(leads_count)s Leads have been created ", leads_count=len(leads_created))
+        else:
+            base_message = _("Aww! No Lead is matching your rule ")
+        #  Add event specific message if event_id is specified
+        if self.event_id:
+            message = base_message + _("for %(event_name)s event.", event_name=self.event_id.name)
+        else:
+            additional_info = _("for ongoing and future events. You can execute it directly from any desired event.")
+            message = base_message + additional_info if leads_created else base_message + _(" for ongoing and future events.") + additional_info
```

Comment:
When dealing with translation, we should avoid to split a sentence in two. So here I think that we should be a bit more verbose to build the different messages :slightly_smiling_face: